### DISCUSSION TITLE
Failed import in RC.101

### DISCUSSION THREAD

**Discussion by TomS-** [directus/directus#9500] [Archive]
Since I've updated to RC.101 I'm not able to build my `SvelteKit` project. The errors I'm receiving are:
```
> Named export 'Directus' not found. The requested module '@directus/sdk' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@directus/sdk';
const { Directus } = pkg;
```
By trying to do a default import I get the following:
```
1: import Directus from '@directus/sdk';
          ^
2: 
3: export const directus = new Directus('https://******.***/');
> 'default' is not exported by node_modules/@directus/sdk/dist/sdk.esm.js, imported by src/lib/directus.js
```

### To Reproduce

Try to build Directus SDK with SvelteKit (Vite). I'm unsure if this effects everything, but judging by the error I assume it would.

### What version of Directus are you using?

RC.101

### What version of Node.js are you using?

14.17.6

### What database are you using?

MySQL 5.7

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying Directus?

Digital Ocean

**Comment by rijkvanzanten** [comment]
It looks like SvelteKit w/ Vite might import the CJS version instead of the ESM build of the SDK. Not sure why though. Any thoughts @nickrum / @joselcvarela ?

**Comment by TomS-** [comment]
It's worth noting that downgrading to RC.100 works, so it's an issue that has changed between RC.100 and RC.101.

**Comment by rijkvanzanten** [comment]
Oh yeah I know exactly what change caused this 😁 (this one: https://github.com/directus/directus/pull/9443) That being said, the question is _why_ that as we (afaik) do the correct thing of pointing to the ESM version on `import` usage: https://github.com/directus/directus/pull/9443/files#diff-77164cc5c071d1488e1cf72c0f91fdd99dd8976edb5004c83198bbd517039572R10-R12

**Comment by nickrum** [comment]
@TomS- Which version of SvelteKit are you using? I can't reproduce this using `@sveltejs/kit@1.0.0-next.193` and a newly scaffolded Svelte project.

**Comment by TomS-** [comment]
> @TomS- Which version of SvelteKit are you using? I can't reproduce this using `@sveltejs/kit@1.0.0-next.193` and a newly scaffolded Svelte project.

I don't get the error until I try to build the project to `adapter-static`.

**Reply by nickrum** [reply]
That makes sense. The issue is actually coming from the SvelteKit SSR server. Should be fixed by #9502.

**Reply by TomS-** [reply]
Thank you @nickrum, however, this is the error I'm now getting

```
> Cannot find module '/www/wwwroot/node_modules/@directus/sdk/index.mjs' imported from /www/wwwroot/.svelte-kit/output/server/app.js
Did you mean to import @directus/sdk/dist/index.js?
```

**Reply by nickrum** [reply]
My bad, the ESM entrypoint wasn't included in the package published to npm. Fixed by #9535.

**Reply by TomS-** [reply]
I've applied your patch @nickrum but I'm now getting:
```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/www/wwwroot/node_modules/@directus/sdk/index.mjs' imported from /www/wwwroot/.svelte-kit/output/server/chunks/__layout-825e2d5f.js
Did you mean to import @directus/sdk/dist/index.js?
```

I replaced the changes back to when it originally worked https://github.com/directus/directus/blob/7e1034d9d4f6847c2cd2b0e43d77ac82f1ee5eea/packages/sdk/package.json#L13-L18 and was able to compile. The changes above do not fix the issue both in `adapter-static` and `adapter-node`.

**Reply by mfeitoza** [reply]
@TomS- same problem here

**Reply by TomS-** [reply]
I wonder why this got moved into discussion, I guess it's technically a bug but can be easily forgotten about as a discussion - leaving many unable to compile their code.

**Reply by rijkvanzanten** [reply]
@TomS- The issue is that it's not so much a bug in our "raw" usage of ESM exports in the package.json files, but an issue in how that interoperates with other packages that import ours in various different ways. In this particular case, the issue arose because svelte-kit seems to rely on the mjs file extension exclusively, as it bundles things differently based on SSR mode vs regular front-end only use. 

There's no one "correct" way of doing these things in NodeJS at the moment (as the industry as a whole is sort of half-switching over to ESM in all sorts of hacky ways), which makes it a little difficult to confidently know how to proceed 😬 I figured that a discussion would be a better place for the time being, as that allows us to have one bigger thread of discussions around integrating with third party packages that might need some special sauce, rather than one bug we can fix that takes care of it all 🤔 happy to hear your thoughts tho!

**Reply by mfeitoza** [reply]
The problem occurs only when add adapter to svelte.config. I tried another ways, but get another type of error.

**Comment by rijkvanzanten** [comment]
(Moving this to a discussion as it feels like a package interoperability issue more than a bug in the SDK we can fix at this point)
